sql/core/src/test/scala/org/apache/spark/sql/execution/SQLViewSuite.scala

```diff
@@ -452,311 +542,96 @@ class SQLViewSuite extends QueryTest with SQLTestUtils with TestHiveSingleton {
-            "`gen_attr_0`, `gen_attr_1` FROM (SELECT `a` AS `gen_attr_0`, `b` AS " +
-            "`gen_attr_1` FROM hive_table) AS gen_subquery_0) AS hive_table")
-        )
-        hiveContext.sessionState.catalog.createTable(view, ignoreIfExists = false)
-        val df = sql("SELECT * FROM old_view")
-        // Check the output rows.
-        checkAnswer(df, Row(1, 2))
-        // Check the output schema.
-        assert(df.schema.sameType(view.schema))
-      }
+  test("correctly resolve a view in a self join") {
+    withView("testView") {
+      sql("CREATE VIEW testView AS SELECT * FROM jt")
+      checkAnswer(
+        sql("SELECT * FROM testView t1 JOIN testView t2 ON t1.id = t2.id ORDER BY t1.id"),
+        (1 to 9).map(i => Row(i, i, i, i)))
     }
   }
 
   test("resolve a view with custom column names") {
-    withTable("testTable") {
-      spark.range(1, 10).selectExpr("id", "id + 1 id1").write.saveAsTable("testTable")
+    withTable("tab1") {
+      spark.range(1, 10).selectExpr("id", "id + 1 id1").write.saveAsTable("tab1")
       withView("testView") {
-        val testView = CatalogTable(
-          identifier = TableIdentifier("testView"),
-          tableType = CatalogTableType.VIEW,
-          storage = CatalogStorageFormat.empty,
-          schema = new StructType().add("x", "long").add("y", "long"),
-          viewText = Some("SELECT * FROM testTable"),
-          properties = Map(CatalogTable.VIEW_DEFAULT_DATABASE -> "default",
-            CatalogTable.VIEW_QUERY_OUTPUT_NUM_COLUMNS -> "2",
-            s"${CatalogTable.VIEW_QUERY_OUTPUT_COLUMN_NAME_PREFIX}0" -> "id",
-            s"${CatalogTable.VIEW_QUERY_OUTPUT_COLUMN_NAME_PREFIX}1" -> "id1"))
-        hiveContext.sessionState.catalog.createTable(testView, ignoreIfExists = false)
+        sql("CREATE VIEW testView(x, y) AS SELECT * FROM tab1")
 
         // Correctly resolve a view with custom column names.
         checkAnswer(sql("SELECT * FROM testView ORDER BY x"), (1 to 9).map(i => Row(i, i + 1)))
 
         // Correctly resolve a view when the referenced table schema changes.
         spark.range(1, 10).selectExpr("id", "id + id dummy", "id + 1 id1")
-          .write.mode(SaveMode.Overwrite).saveAsTable("testTable")
+          .write.mode(SaveMode.Overwrite).saveAsTable("tab1")
         checkAnswer(sql("SELECT * FROM testView ORDER BY x"), (1 to 9).map(i => Row(i, i + 1)))
 
         // Throw an AnalysisException if the column name is not found.
         spark.range(1, 10).selectExpr("id", "id + 1 dummy")
-          .write.mode(SaveMode.Overwrite).saveAsTable("testTable")
+          .write.mode(SaveMode.Overwrite).saveAsTable("tab1")
         intercept[AnalysisException](sql("SELECT * FROM testView"))
       }
     }
   }
 
   test("resolve a view when the dataTypes of referenced table columns changed") {
-    withTable("testTable") {
-      spark.range(1, 10).selectExpr("id", "id + 1 id1").write.saveAsTable("testTable")
+    withTable("tab1") {
```

Comment:
For this test case I think we'd better create a new table, because we would overwrite the table in the test case, if we use `jt` then we have to recover the table content and that may be error prone.